plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergOrcFileWriter.java

```diff
@@ -261,10 +277,33 @@ private static int getIcebergId(OrcType orcColumn)
         private ByteBuffer min;
         private ByteBuffer max;
 
-        private IcebergMinMax(org.apache.iceberg.types.Type type, Object min, Object max)
+        private IcebergMinMax(org.apache.iceberg.types.Type type, Object min, Object max, MetricsModes.MetricsMode metricsMode)
         {
-            this.min = Conversions.toByteBuffer(type, min);
-            this.max = Conversions.toByteBuffer(type, max);
+            if (metricsMode instanceof MetricsModes.Full) {
+                this.min = Conversions.toByteBuffer(type, min);
+                this.max = Conversions.toByteBuffer(type, max);
+            }
+            else if (metricsMode instanceof MetricsModes.Truncate) {
+                MetricsModes.Truncate truncateMode = (MetricsModes.Truncate) metricsMode;
+                int truncateLength = truncateMode.length();
+                switch (type.typeId()){
```

Comment:
```suggestion
                switch (type.typeId()) {
```

cc @ksobolew , why not require whitespace before `{`?

Reply:
Checkstyle didn't complain? Oh, it didn't indeed... Looks like a gap in the configuration

Reply:
...or it's a bug in checkstyle. Whitespace before `{` is not allowed by our configuration, and checkstyle complains about it *except* after a `switch (...)`.